src/main/java/org/example/Main.java

```diff
@@ -17,83 +13,117 @@
+  public static void main(String[] args) throws Exception {
+
+    String apiUrl = "https://api.fda.gov/drug/drugsfda.json?&limit=1000";
+
+    URL url = new URL(apiUrl);
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+    connection.setRequestMethod("GET");
+
+    int responseCode = connection.getResponseCode();
+    if (responseCode == HttpURLConnection.HTTP_OK) {
+      BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+      String inputLine;
+      StringBuffer response = new StringBuffer();
+      while ((inputLine = in.readLine()) != null) {
+        response.append(inputLine);
+      }
+      in.close();
+
+      JsonObject jsonObject = new Gson().fromJson(response.toString(), JsonObject.class);
+      JsonArray results = jsonObject.getAsJsonArray("results");
+
+      Class.forName("org.postgresql.Driver");
+
+      Connection jDbCconnection =
+          DriverManager.getConnection("jdbc:postgresql://localhost:5432/dex");
+      PreparedStatement statement =
+          jDbCconnection.prepareStatement(
+              "INSERT INTO medication_with_uuid (brand_name, strength, dosage_form) VALUES (?,?,?)");
+
+      List<String> brandNameList = new ArrayList<>();
+      List<Medication> medications = new ArrayList<>();
+      for (JsonElement resultElement : results) {
+        JsonObject result = resultElement.getAsJsonObject();
+        // products may be null
+        Optional<JsonArray> maybeProducts = Optional.ofNullable(result.getAsJsonArray("products"));
+        if (maybeProducts.isEmpty()) {
+          continue;
+        }
+        // if we're here that means products are good to go
+        JsonArray products = maybeProducts.get();
+
+        Optional<String> maybeBrandName =
+            Optional.ofNullable(
+                products
+                    .getAsJsonArray()
+                    .get(0)
+                    .getAsJsonObject()
+                    .getAsJsonPrimitive("brand_name")
+                    .getAsString());
+
+        Optional<String> maybeStrength =
+            Optional.ofNullable(
+                products
+                    .getAsJsonArray()
+                    .get(0)
+                    .getAsJsonObject()
+                    .getAsJsonArray("active_ingredients")
+                    .getAsJsonArray()
+                    .get(0)
+                    .getAsJsonObject()
+                    .getAsJsonPrimitive("strength")
+                    .getAsString());
+
+        Optional<String> maybe_dosage_form =
+            Optional.ofNullable(
+                products
+                    .getAsJsonArray()
+                    .get(0)
+                    .getAsJsonObject()
+                    .getAsJsonPrimitive("dosage_form")
+                    .getAsString());
+        if (maybeBrandName.isEmpty() || maybeStrength.isEmpty() || maybe_dosage_form.isEmpty()) {
+          continue;
         }
 
```

Comment:
Will this skip medications that have some null values and just add the medications with all values to the database? Just wondering if some medications will be missed and if it will matter.

Reply:
this will skip all results that don't have a product array, brand name, strength and dosage... like all four. So technically yes, as this code stands some medications will be missed as i unwrap the result object.  In this case... out of 1000 results... my table now has about 330 or so rows. 1000 seems to be the most i can do without things breaking (would probably need to test that further). I'm pretty sure, to get the most results possible... i would probably have to add more checks in between the conditions i've already set... it's not impossible but it may not be a priority?? Josh says 330 medications is good enough to start with. My biggest concern was making sure i wasn't putting null values into the table.